### PR TITLE
#200 Docker on the agent's side

### PR DESCRIPTION
**Pull request by vincent-zurczak**

### CODE REVIEW

File: core/roboconf-core/src/main/java/net/roboconf/core/ErrorCode.java [hunk 1]
```diff
@@ -124,6 +124,8 @@
 	RM_AMBIGUOUS_OVERRIDING( ErrorLevel.SEVERE, ErrorCategory.RUNTIME_MODEL, "A variable is exported in the instance but is ambiguously resolved in its component." ),
 	RM_INVALID_EXTERNAL_EXPORT( ErrorLevel.SEVERE, ErrorCategory.RUNTIME_MODEL, "The application exports a variable that does not exist in the graph." ),
 	RM_ALREADY_DEFINED_EXTERNAL_EXPORT( ErrorLevel.SEVERE, ErrorCategory.RUNTIME_MODEL, "The application exports a variable more than once. This is probably a copy/paste error." ),
+	RM_INVALID_RANDOM_KIND( ErrorLevel.SEVERE, ErrorCategory.RUNTIME_MODEL, "A component exports a variable whose value is generated randmly by Roboconf, but its type is invalid." ),
```
Comment (gibello) [inline core/roboconf-core/src/main/java/net/roboconf/core/ErrorCode.java]: Randomly (misspelled)

File: core/roboconf-dm/src/main/java/net/roboconf/dm/internal/api/impl/RandomMngrImpl.java [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+		}
+	}
+
+
+	// Port Management
+
+	private static final int RABBITMQ_PORT = 35197;
+	private static final int PORT_MIN = 10000;
+	private static final int PORT_MAX = 65500;
+
```
Comment (gibello) [inline core/roboconf-dm/src/main/java/net/roboconf/dm/internal/api/impl/RandomMngrImpl.java]: Forbidden ports are a problem: a typical /etc/services file often lists ports above 10000, and some apps may use such ports without any declaration in /etc/services (eg. RabbitMQ, but it is not the only one).
So, how can we manage "forbidden" ports so there is no conflict ? Declare exceptions somewhere ??

File: core/roboconf-core/src/main/java/net/roboconf/core/ErrorCode.java (hunk 1 above)
Comment (vincent-zurczak) [inline core/roboconf-core/src/main/java/net/roboconf/core/ErrorCode.java]: Fixed locally. I will push it with next commits (no need to waste one for this).

File: core/roboconf-dm/src/main/java/net/roboconf/dm/internal/api/impl/RandomMngrImpl.java (hunk 2 above)
Comment (vincent-zurczak) [inline core/roboconf-dm/src/main/java/net/roboconf/dm/internal/api/impl/RandomMngrImpl.java]: Indeed. To be managed as another issue.
We can add a list of forbidden ports in the DM's configuration (e.g. in the REST API or in its OSGi configuration).

**Pull request merged**